File: Labs/Lab1/Lab1.py
class Solution(object):
    def twoSum(self, nums, target):     #inputs: array nums, int target

        hashbrown = {}                              #hashmap
        for i, currentEntry in enumerate(nums):     #i is index (count from 0), currentEntry is currentEntry, use enumerate to add to hashmap

            if ((target - currentEntry) in hashbrown) : #if target-currentEntry is already in the hashmap
                return(hashbrown.get(target-currentEntry), i)  #return the value of target-currentEntry and index of currentEntry
            
            hashbrown.update({currentEntry: i})     #else, add the currentEntry and the index
        
        return("none :(")       #else, return nothing :(


test = Solution()
print(test.twoSum([1, 2, 3, 4, 5], 9))
print(test.twoSum([123, 32, 2, 34123, 44], 3))
print(test.twoSum([2, 3, 4, 6], 10))

File: Labs/Lab1/Lab1.java
import java.util.HashMap;    //import lol

public class Solution {

    public static int[] twoSum(int[] nums, int target) {
            
            HashMap<Integer, Integer> hashbrown = new HashMap<>(); //hashmap
    
            for (int i = 0; i < nums.length; i++) {     //i is index (count from 0) idk how to enumerate so ig its c++ methods now
    
                int currentEntry = nums[i];    //currentEntry is current entry as according to index, cycle thru
                if (hashbrown.containsKey(target - currentEntry)) { //if target-currentEntry is already in the hashmap
                    
                    return new int[] {hashbrown.get(target-currentEntry), i};  //return the value of target-currentEntry and index of currentEntry
                }
                hashbrown.put(currentEntry, i);     //else, add the currentEntry and the index
            }
    
            return new int[]{};       //else, return nothing :(
        }

    public static void main(String[] args) {
        
        int[] nums = {1, 2, 3, 4};
        int target = 5;
        int[] result = twoSum(nums, target);
        
        System.out.println(result[0] + ", " + result[1]);
    }
}
